nebula/tests/transcript/test_queue_worker.py

```
@@ -110,3 +110,199 @@ async def fake_light_phase(job_id, req, transcription, video_path, audio_path, u
+    assert "job2" not in remaining_jobs
+    assert "job3" in remaining_jobs
+
+
+@pytest.mark.anyio
+async def test_cancel_job_during_processing_stops_it(monkeypatch):
+    """Test that cancelling a job during processing stops it and cleans up."""
+    monkeypatch.setattr(qw, "_job_queue", asyncio.Queue())
+
+    cleanup_called = []
+
+    async def fake_heavy_pipeline(job_id, req):
```

Ruff (0.14.6)

Unused function argument: `req`

(ARG001)